### PR TITLE
feat: run sub workflows in the same process

### PR DESCRIPTION
**Pull request by MasterPtato**

### CODE REVIEW

**Comment by MasterPtato** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/789?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#789** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/789?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#788** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/788?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#787** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/787?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#785** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/785?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#784** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/784?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#783** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/783?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#780** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/780?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MasterPtato and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by AngelOnFira — APPROVED** [review]

File: lib/chirp-workflow/core/src/ctx/workflow.rs [hunk 1]
```diff
@@ -373,11 +373,66 @@ impl WorkflowCtx {
...
+			registry: self.registry.clone(),
+			db: self.db.clone(),
+
+			conn: self
+				.conn
+				.wrap(Uuid::new_v4(), self.ray_id, I::Workflow::name()),
+
+			event_history: self.event_history.clone(),
+
+			// TODO: This is redundant with the deserialization in `workflow.run` in the registry
```
Comment (NathanFlurry) [inline lib/chirp-workflow/core/src/ctx/workflow.rs:414]: ticket

File: lib/chirp-workflow/core/src/ctx/workflow.rs [hunk 2]
```diff
@@ -373,11 +373,66 @@ impl WorkflowCtx {
...
-		Ok(output)
+		// Lookup workflow
+		let Ok(workflow) = self.registry.get_workflow(I::Workflow::name()) else {
+			tracing::warn!(
+				id=%self.workflow_id,
+				name=%I::Workflow::name(),
+				"sub workflow not found in current registry",
+			);
+			
+			// TODO: If a sub workflow is dispatched, then the worker is updated to include the sub worker
```
Comment (NathanFlurry) [inline lib/chirp-workflow/core/src/ctx/workflow.rs:384]: ticket

File: lib/chirp-workflow/core/src/ctx/workflow.rs [hunk 3]
```diff
@@ -373,11 +373,66 @@ impl WorkflowCtx {
...
+				.collect(),
+			location_idx: 0,
+		};
+
+		self.location_idx += 1;
+
+		// Run workflow
+		let output = (workflow.run)(&mut ctx).await?;
+
+		// TODO: Get rid of this step, this is redundant
```
Comment (NathanFlurry) [inline lib/chirp-workflow/core/src/ctx/workflow.rs:431]: ticket

**Review by NathanFlurry — CHANGES_REQUESTED** [review]

**Review by NathanFlurry — APPROVED** [review]

**Review by NathanFlurry — APPROVED** [review]

**Pull request merged**